freeipa/src/test/java/com/sequenceiq/freeipa/converter/instance/InstanceGroupRequestToInstanceGroupConverterTest.java

```diff
@@ -276,17 +278,91 @@ void convertTestDefaultTemplateConversionWithEncryptionKey() {
                 detailedEnvironmentResponse,
-                createAndGetCloudArgsForIgCoverterMap(null, "dummyEncryptionKey"));
+                createAndGetCloudArgsForIgCoverterMap(null, "dummyEncryptionKey", null));
 
         assertThat(result).isNotNull();
         assertThat(result.getTemplate()).isSameAs(template);
     }
 
-    EnumMap<CloudArgsForIgConverter, String> createAndGetCloudArgsForIgCoverterMap(String diskEncryptionSetId, String encryptionKey) {
+    @Test
+    void convertTestTemplateConversionWithAwsEncryptionKey() {
+        InstanceGroupRequest request = new InstanceGroupRequest();
+
+        InstanceTemplateRequest instanceTemplateRequest = mock(InstanceTemplateRequest.class);
+        request.setInstanceTemplateRequest(instanceTemplateRequest);
+        Template template = mock(Template.class);
+
+        Stack stack = new Stack();
+        stack.setAccountId(ACCOUNT_ID);
+        stack.setCloudPlatform(MOCK.name());
+        stack.setName(NAME);
+
+        FreeIpaServerRequest freeIpaServerRequest = new FreeIpaServerRequest();
+        freeIpaServerRequest.setHostname(HOSTNAME);
+        freeIpaServerRequest.setDomain(DOMAINNAME);
+
+        NetworkRequest networkRequest = new NetworkRequest();
+
+        DetailedEnvironmentResponse detailedEnvironmentResponse = new DetailedEnvironmentResponse();
+
+        when(templateConverter.convert(instanceTemplateRequest, MOCK, ACCOUNT_ID, null, null, "dummyAwsEncryptionKey")).thenReturn(template);
+
+        InstanceGroup result = underTest.convert(
+                request,
+                networkRequest,
+                ACCOUNT_ID,
+                stack,
+                freeIpaServerRequest,
+                detailedEnvironmentResponse,
+                createAndGetCloudArgsForIgCoverterMap(null, null, "dummyAwsEncryptionKey"));
+
+        assertThat(result).isNotNull();
+        assertThat(result.getTemplate()).isSameAs(template);
+    }
+
+    @Test
+    void convertTestDefaultTemplateConversionWithAwsEncryptionKey() {
+        InstanceGroupRequest request = new InstanceGroupRequest();
+
+        InstanceTemplateRequest instanceTemplateRequest = mock(InstanceTemplateRequest.class);
+        request.setInstanceTemplateRequest(instanceTemplateRequest);
+        Template template = mock(Template.class);
+
+        Stack stack = new Stack();
+        stack.setAccountId(ACCOUNT_ID);
+        stack.setCloudPlatform(MOCK.name());
+        stack.setName(NAME);
+
+        FreeIpaServerRequest freeIpaServerRequest = new FreeIpaServerRequest();
+        freeIpaServerRequest.setHostname(HOSTNAME);
+        freeIpaServerRequest.setDomain(DOMAINNAME);
+
+        NetworkRequest networkRequest = new NetworkRequest();
+
+        DetailedEnvironmentResponse detailedEnvironmentResponse = new DetailedEnvironmentResponse();
+
+        when(templateConverter.convert(instanceTemplateRequest, MOCK, ACCOUNT_ID, null, null, "dummyAwsEncryptionKey")).thenReturn(template);
```

Comment:
To test the default we should not pass "dummyAwsEncryptionKey" right?

Reply:
Yes, I guess this needs to be fixed for Azure and Gcp as well as I see two test cases with different names doing exactly same thing in both clouds as well.